node/src/blockchain/blockchain_interface/blockchain_interface_web3/lower_level_interface_web3.rs

```diff
@@ -1,395 +1,762 @@
+            .get_transaction_receipt(tx_hash)
+            .wait()
+            .unwrap_err();
+        let error_message = if let BlockchainError::QueryFailed(em) = actual_error {
+            em
+        } else {
+            panic!("Expected BlockchainError::QueryFailed(msg)");
+        };
+        assert_string_contains(
+            error_message.as_str(),
+            "Transport error: Error(Connect, Os { code: ",
+        );
+        assert_string_contains(
+            error_message.as_str(),
+            ", kind: ConnectionRefused, message: ",
+        );
     }
 
     #[test]
-    fn low_interface_web3_handles_latest_null_block_number_error() {
-        let prepare_params_arc = Arc::new(Mutex::new(vec![]));
-        let transport = TestTransport::default()
-            .prepare_params(&prepare_params_arc)
-            .send_result(Value::Null);
-        let subject = make_subject(transport, TEST_DEFAULT_CHAIN);
+    fn transaction_receipt_batch_works() {
+        let port = find_free_port();
+        let tx_hash_1 =
+            H256::from_str("a128f9ca1e705cc20a936a24a7fa1df73bad6e0aaf58e8e6ffcc154a7cff6e0e")
+                .unwrap();
+        let tx_hash_2 =
+            H256::from_str("a128f9ca1e705cc20a936a24a7fa1df73bad6e0aaf58e8e6ffcc154a7cff6e0f")
+                .unwrap();
+        let tx_hash_3 =
+            H256::from_str("a128f9ca1e705cc20a936a24a7fa1df73bad6e0aaf58e8e6ffcc154a7cff6e0a")
+                .unwrap();
+        let tx_hash_4 =
+            H256::from_str("a128f9ca1e705cc20a936a24a7fa1df73bad6e0aaf58e8e6ffcc154a7cff6e0b")
+                .unwrap();
+        let tx_hash_vec = vec![tx_hash_1, tx_hash_2, tx_hash_3, tx_hash_4];
+        let block_hash =
+            H256::from_str("6d0abccae617442c26104c2bc63d1bc05e1e002e555aec4ab62a46e826b18f18")
+                .unwrap();
+        let block_number = U64::from_str("b0328d").unwrap();
+        let cumulative_gas_used = U256::from_str("60ef").unwrap();
+        let gas_used = U256::from_str("60ef").unwrap();
+        let status = U64::from(0);
+        let logs_bloom = H2048::from_str("00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000800000000000000000000000000000000000").unwrap();
+        let tx_receipt_response = ReceiptResponseBuilder::default()
+            .transaction_hash(tx_hash_1)
+            .block_hash(block_hash)
+            .block_number(block_number)
+            .cumulative_gas_used(cumulative_gas_used)
+            .gas_used(gas_used)
+            .status(status)
+            .logs_bloom(logs_bloom)
+            .build();
```

Comment:
Yeah, this is quite a monster, too big piece of information where just a little fraction is useful and would likely ever be, and this builder will never have a good usecase of which reason it shouldn't exist.